apis/v1alpha2/validation/httproute.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+var (
+	// repeatableHTTPRouteFilters are filter types that can are allowed to be
+	// repeated multiple times in a rule.
+	repeatableHTTPRouteFilters = []gatewayv1a2.HTTPRouteFilterType{
+		gatewayv1a2.HTTPRouteFilterExtensionRef,
+	}
+)
+
+// ValidateHTTPRoute validates HTTPRoute according to the Gateway API specification.
+// For additional details of the HTTPRoute spec, refer to:
+// https://gateway-api.sigs.k8s.io/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRoute
+func ValidateHTTPRoute(route *gatewayv1a2.HTTPRoute) field.ErrorList {
```

Comment:
Looking at `HTTPRoute` 
1. there are plenty of slices. Are we ok with them all being zero lengths?
2. some data like `HTTPRouteFilterType` which is an enum but not validated. Is that expected?

Reply:
> 1. there are plenty of slices. Are we ok with them all being zero lengths?

I think for the most part yes. That does result in some unfortunate possibilities, including an HTTPRoute with no BackendRefs. We allow this because it's possible to specify a terminating filter such as redirect, in which case a backend would not make sense. This is something we could try to catch with the validating webhook, but even then a webhook is not failproof, there's no guarantee that webhook validation won't be bypassed.

> 2. some data like HTTPRouteFilterType which is an enum but not validated. Is that expected?

I have no idea how we missed that, thanks for catching that one! #772 covers a lot of missing validation but also missed that one.

Reply:
Actually, I was wrong, HTTPRouteFilterType does enum validation: https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/httproute_types.go#L551. I think that combined with the other validation improvements in #772 is sufficient to resolve this, feel free to unresolve if I missed anything though.